var/spack/repos/builtin/packages/suite-sparse/package.py

```diff
@@ -75,47 +83,28 @@ def install(self, spec, prefix):
             # completely disabled. See
             # [SuiteSparse/SuiteSparse_config/SuiteSparse_config.mk] for more.
             'CUDA=no',
-            'CUDA_PATH={0}'.format(
-                spec['cuda'].prefix if '+cuda' in spec else ''
-            ),
-            'CFOPENMP={0}'.format(self.compiler.openmp_flag)
-        ])
-
-        if '+pic' in spec:
-            make_args.extend([
-                'CFLAGS={0}'.format(self.compiler.pic_flag),
-                'FFLAGS={0}'.format(self.compiler.pic_flag)
-            ])
+            'CUDA_PATH=%s' % (spec['cuda'].prefix if '+cuda' in spec else ''),
+            'CFOPENMP=%s' % omp_flag,
+            'CFLAGS=-O3 -fexceptions -fno-common %s' % pic_flag,
+            'F77FLAGS=%s' % pic_flag,
```

Comment:
There is no `FFLAGS` set by `make config` for `suite-sparse` versions 4.5.1 and higher.

```
oxberry1 > ~ > Downloads > SuiteSparse-4.5.1 > make config

----------------------------------------------------------------
SuiteSparse package compilation options:
----------------------------------------------------------------

SuiteSparse Version:      4.5.1
SuiteSparse top folder:   /Users/oxberry1/Downloads/SuiteSparse-4.5.1
Package:                  LIBRARY=         PackageNameWillGoHere
Version:                  VERSION=         x.y.z
SO version:               SO_VERSION=      x
System:                   UNAME=           Darwin
Install directory:        INSTALL=         /Users/oxberry1/Downloads/SuiteSparse-4.5.1
Install libraries in:     INSTALL_LIB=     /Users/oxberry1/Downloads/SuiteSparse-4.5.1/lib
Install include files in: INSTALL_INCLUDE= /Users/oxberry1/Downloads/SuiteSparse-4.5.1/include
Install documentation in: INSTALL_DOC=     /Users/oxberry1/Downloads/SuiteSparse-4.5.1/share/doc/suitesparse-4.5.1
Optimization level:       OPTIMIZATION=    -O3
BLAS library:             BLAS=            -framework Accelerate
LAPACK library:           LAPACK=          -framework Accelerate
Intel TBB library:        TBB=
Other libraries:          LDLIBS=          -lm
static library:           AR_TARGET=       PackageNameWillGoHere.a
shared library (full):    SO_TARGET=       PackageNameWillGoHere.x.y.z.dylib
shared library (main):    SO_MAIN=         PackageNameWillGoHere.x.dylib
shared library (short):   SO_PLAIN=        PackageNameWillGoHere.dylib
shared library options:   SO_OPTS=         -L/Users/oxberry1/Downloads/SuiteSparse-4.5.1/lib -dynamiclib -compatibility_version x -current_version x.y.z -shared -undefined dynamic_lookup
shared library name tool: SO_INSTALL_NAME= install_name_tool -id
ranlib, for static libs:  RANLIB=          ranlib
static library command:   ARCHIVE=         ar rv
copy file:                CP=              cp -f
move file:                MV=              mv -f
remove file:              RM=              rm -f
pretty (for Tcov tests):  PRETTY=          grep -v "^#" | indent -bl -nce -bli0 -i4 -sob -l120
C compiler:               CC=              cc
C++ compiler:             CXX=             c++
CUDA compiler:            NVCC=            echo
CUDA root directory:      CUDA_PATH=
OpenMP flags:             CFOPENMP=
C/C++ compiler flags:     CF=                 -O3 -fexceptions -fPIC -fno-common
LD flags:                 LDFLAGS=         -L/Users/oxberry1/Downloads/SuiteSparse-4.5.1/lib
Fortran compiler:         F77=             f77
Fortran flags:            F77FLAGS=
Intel MKL root:           MKLROOT=
Auto detect Intel icc:    AUTOCC=          yes
UMFPACK config:           UMFPACK_CONFIG=
CHOLMOD config:           CHOLMOD_CONFIG=
SuiteSparseQR config:     SPQR_CONFIG=
CUDA library:             CUDART_LIB=
CUBLAS library:           CUBLAS_LIB=
METIS and CHOLMOD/Partition configuration:
Your METIS library:       MY_METIS_LIB=
Your metis.h is in:       MY_METIS_INC=
METIS is used via the CHOLMOD/Partition module, configured as follows.
If the next line has -DNPARTITION then METIS will not be used:
CHOLMOD Partition config:
CHOLMOD Partition libs:    -lccolamd -lcamd /Users/oxberry1/Downloads/SuiteSparse-4.5.1/lib/libmetis.dylib
CHOLMOD Partition include: -I/Users/oxberry1/Downloads/SuiteSparse-4.5.1/CCOLAMD/Include -I/Users/oxberry1/Downloads/SuiteSparse-4.5.1/CAMD/Include -I/Users/oxberry1/Downloads/SuiteSparse-4.5.1/metis-5.1.0/include
```

However, being paranoid:

```
 oxberry1 > ~ > Downloads > SuiteSparse-4.5.1 > find . -name Makefile | xargs grep FFLAGS
./CHOLMOD/Demo/Makefile:        $(F77) $(FFLAGS) -o readhb readhb.f
./CHOLMOD/Demo/Makefile:        $(F77) $(FFLAGS) -O -o readhb2 readhb2.f
./CHOLMOD/Demo/Makefile:        $(F77) $(FFLAGS) -O -o reade reade.f
 oxberry1 > ~ > Downloads > SuiteSparse-4.5.1 > find . -name Makefile | xargs grep F77FLAGS
./AMD/Demo/Makefile:    $(F77) $(F77FLAGS) -o amd_f77demo amd_f77demo.f ../Lib/libamdf77.a \
./AMD/Demo/Makefile:    $(F77) $(F77FLAGS) -o amd_f77simple amd_f77simple.f \
./AMD/Demo/Makefile:    $(F77) $(F77FLAGS) -o amd_f77cross amd_f77cross.f amd_f77wrapper.o \
./AMD/Lib/Makefile:     $(F77) $(F77FLAGS) -c ../Source/amd.f -o amd.o
./AMD/Lib/Makefile:     $(F77) $(F77FLAGS) -c ../Source/amdbar.f -o amdbar.o
./SuiteSparse_config/xerbla/Makefile:    COMPILE = $(F77) $(F77FLAGS) -c xerbla.f
./UMFPACK/Demo/Makefile:        $(F77) $(F77FLAGS) -o readhb readhb.f $(F77LIB)
./UMFPACK/Demo/Makefile:        $(F77) $(F77FLAGS) -o readhb_size readhb_size.f $(F77LIB)
./UMFPACK/Demo/Makefile:        $(F77) $(F77FLAGS) -o readhb_nozeros readhb_nozeros.f $(F77LIB)
./UMFPACK/Demo/Makefile:        $(F77) $(F77FLAGS) -o umf4hb umf4hb.f umf4_f77wrapper.o $(LIBS)
./UMFPACK/Demo/Makefile:        $(F77) $(F77FLAGS) -o umf4zhb umf4zhb.f umf4_f77zwrapper.o $(LIBS)
./UMFPACK/Demo/Makefile:        $(F77) $(F77FLAGS) -o umf4hb64 umf4hb64.f umf4_f77wrapper64.o $(LIBS)
```
so both `FFLAGS` and `F77FLAGS` must be set.